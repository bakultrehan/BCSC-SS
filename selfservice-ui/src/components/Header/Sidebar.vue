/** * Header of app */

<template>
  <v-card class="mx-auto">
    <v-navigation-drawer app clipped fixed height="100%" permanent>
      <!-- v-model="drawer" -->
      <v-list dark>
        <v-list-item>
          <v-list-item-content>
            <v-list-item-title class="title">Self Service</v-list-item-title>
            <!-- <v-list-item-subtitle>
              subtext
            </v-list-item-subtitle>-->
          </v-list-item-content>
        </v-list-item>

        <v-divider></v-divider>

        <v-list dense nav>
          <v-list-item v-for="item in items" :key="item.title" link>
            <v-list-item-icon>
              <v-icon>{{ item.icon }}</v-icon>
            </v-list-item-icon>

            <v-list-item-content>
              <v-list-item-title :to="item.title">
                <Link :to="item.link">{{ item.title }}</Link>
              </v-list-item-title>
            </v-list-item-content>
          </v-list-item>
        </v-list>
      </v-list>
    </v-navigation-drawer>
  </v-card>
</template>

<script lang="ts">
import { Component, Vue, Prop } from 'vue-property-decorator';
import { Getter, namespace, Action } from 'vuex-class';
import Link from '@/Atomic/Link/Link.vue';

const KeyCloakModule = namespace('KeyCloakModule');

@Component({
  components: {
    Link
  }
})
export default class Sidebar extends Vue {
  @Prop() private drawer!: boolean;
  private items: any = [
    { title: 'Dashboard', icon: 'mdi-view-dashboard', link: '/' },
    { title: 'Create Project', icon: 'mdi-plus-circle', link: '/project/info' },
    { title: 'About', icon: 'mdi-help-box', link: '/about' }
  ];
}
</script>

<style lang="scss" scoped>
@import './../../assets/styles/theme.scss';

.v-navigation-drawer {
  text-align: left;
  background-color: $BCgovBlue5 !important;
}
</style>
