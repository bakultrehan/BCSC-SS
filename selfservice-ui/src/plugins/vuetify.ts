import Vue from 'vue';
import Vuetify from 'vuetify/lib';

Vue.use(Vuetify);

export default new Vuetify({
  icons: {
    iconfont: 'mdi',
  },
  theme: {
    themes: {
      light: {
        error: '#D8292F',
        success: '#2E8540',
        warning: '#FFC107',
      },
    },
    options: {
      cspNonce: 'cLv1G2y8KtMdR',
    },
  },
});
