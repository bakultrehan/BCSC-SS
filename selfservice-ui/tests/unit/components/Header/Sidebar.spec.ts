/* tslint:disable */
import { shallowMount, createLocalVue } from '@vue/test-utils';
import Sidebar from '@/components/Header/Sidebar.vue';
import vuetify from 'vuetify';

describe('Sidebar.vue', () => {
  let wrapper: any;
  beforeEach(() => {
    const localVue = createLocalVue();
    localVue.use(vuetify);

    wrapper = shallowMount(Sidebar, {
      localVue,
      mocks: { $t: jest.fn(() => {}) } // tslint:disable-line
    });
  });

  it('renders props when passed', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
