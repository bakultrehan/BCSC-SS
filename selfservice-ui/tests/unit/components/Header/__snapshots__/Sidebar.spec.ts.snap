// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar.vue renders props when passed 1`] = `
<v-card-stub
  class="mx-auto"
  loaderheight="4"
  tag="div"
>
  <v-navigation-drawer-stub
    app="true"
    clipped="true"
    fixed="true"
    height="100%"
    minivariantwidth="56"
    mobilebreakpoint="1264"
    permanent="true"
    src=""
    tag="nav"
    width="256"
  >
    <v-list-stub
      dark="true"
      tag="div"
      tile="true"
    >
      <v-divider-stub />
       
      <v-list-stub
        dense="true"
        nav="true"
        tag="div"
        tile="true"
      >
        <v-list-item-stub
          activeclass=""
          link="true"
          tag="div"
        >
          <v-list-item-icon-stub>
            <v-icon-stub>
              mdi-home
            </v-icon-stub>
          </v-list-item-icon-stub>
           
          <v-list-item-content-stub>
            <v-list-item-title-stub
              to="Home"
            >
              <link-stub
                to="/"
              >
                Home
              </link-stub>
            </v-list-item-title-stub>
          </v-list-item-content-stub>
        </v-list-item-stub>
        <v-list-item-stub
          activeclass=""
          link="true"
          tag="div"
        >
          <v-list-item-icon-stub>
            <v-icon-stub>
              mdi-view-dashboard
            </v-icon-stub>
          </v-list-item-icon-stub>
           
          <v-list-item-content-stub>
            <v-list-item-title-stub
              to="Dashboard"
            >
              <link-stub
                to="/dashboard"
              >
                Dashboard
              </link-stub>
            </v-list-item-title-stub>
          </v-list-item-content-stub>
        </v-list-item-stub>
        <v-list-item-stub
          activeclass=""
          link="true"
          tag="div"
        >
          <v-list-item-icon-stub>
            <v-icon-stub>
              mdi-plus-circle
            </v-icon-stub>
          </v-list-item-icon-stub>
           
          <v-list-item-content-stub>
            <v-list-item-title-stub
              to="Create Project"
            >
              <link-stub
                to="/project/info"
              >
                Create Project
              </link-stub>
            </v-list-item-title-stub>
          </v-list-item-content-stub>
        </v-list-item-stub>
        <v-list-item-stub
          activeclass=""
          link="true"
          tag="div"
        >
          <v-list-item-icon-stub>
            <v-icon-stub>
              mdi-help-box
            </v-icon-stub>
          </v-list-item-icon-stub>
           
          <v-list-item-content-stub>
            <v-list-item-title-stub
              to="About"
            >
              <link-stub
                to="/about"
              >
                About
              </link-stub>
            </v-list-item-title-stub>
          </v-list-item-content-stub>
        </v-list-item-stub>
      </v-list-stub>
    </v-list-stub>
  </v-navigation-drawer-stub>
</v-card-stub>
`;
